### PR TITLE
[Docs] `sort-comp`: add class component examples

### PR DESCRIPTION
**Pull request by maurer2**
Hello, I updated the examples for sort-comp, that were still using `createClass` with class component examples.

Cheers

### CODE REVIEW

File: docs/rules/sort-comp.md [hunk 1]
```diff
@@ -16,23 +16,23 @@ The default configuration ensures that the following order must be followed:
...
 
 Examples of **correct** code for this rule:
 
 ```jsx
-var Hello = createReactClass({
-  displayName : 'Hello',
-  render: function() {
+class Hello extends React.Component {
+  static displayName = 'Hello';
+  render(){
```
Comment (ljharb) [inline docs/rules/sort-comp.md:32]:
```suggestion
  render () {
```

File: docs/rules/sort-comp.md [hunk 2]
```diff
@@ -120,23 +120,23 @@ For example, if you want to place your event handlers (`onClick`, `onSubmit`, et
 Examples of **incorrect** code for this rule, with the above configuration:
 
 ```jsx
-var Hello = createReactClass({
-  render: function() {
+class Hello extends React.Component {
+  render(){
```
Comment (ljharb) [inline docs/rules/sort-comp.md:124]:
```suggestion
  render () {
```

File: docs/rules/sort-comp.md [hunk 3]
```diff
@@ -120,23 +120,23 @@ For example, if you want to place your event handlers (`onClick`, `onSubmit`, et
...
     return <div>Hello</div>;
-  },
-  onClick: function() {}
-});
+  }
+  onClick = () => {}
```
Comment (ljharb) [inline docs/rules/sort-comp.md:127]: functions should never be stored in class fields
```suggestion
  onClick = this.onClick.bind(this);
  onClick() {}
```

File: docs/rules/sort-comp.md [hunk 4]
```diff
@@ -120,23 +120,23 @@ For example, if you want to place your event handlers (`onClick`, `onSubmit`, et
...
 ```
 
 Examples of **correct** code for this rule, with the above configuration:
 
 ```jsx
-var Hello = createReactClass({
-  onClick: function() {},
-  render: function() {
+class Hello extends React.Component {
+  onClick = () => {}
```
Comment (ljharb) [inline docs/rules/sort-comp.md:135]: same here

**Review by ljharb — CHANGES_REQUESTED** [review]
Rather than replacing these examples, let's add them so we have both

**Comment by maurer2** [comment]
Okay, I restored the `createClass` examples and fixed the formatting issues.
Cheers

File: docs/rules/sort-comp.md [hunk 5]
```diff
@@ -24,6 +24,15 @@ var Hello = createReactClass({
 });
 ```
 
+```jsx
+class Hello extends React.Component {
+  render () {
```
Comment (ljharb) [inline docs/rules/sort-comp.md:29]:
```suggestion
  render() {
```
whoops, sorry, i think i spaced it wrong in my original suggestion. same throughout.

**Comment by maurer2** [comment]
Implemented the onClick changes, but I think `onClick = this.onClick.bind(this);` needs to be in the constructor. Public class fields should also work as an alternative.

**Review by ljharb — APPROVED** [review]
Thanks!

**Pull request merged**